tests/components/homematicip_cloud/test_init.py

```diff
@@ -10,9 +10,7 @@
 
 async def test_config_with_accesspoint_passed_to_config_entry(hass):
     """Test that config for a accesspoint are loaded via config entry."""
-    with patch.object(hass, "config_entries") as mock_config_entries, patch.object(
-        hmipc, "configured_haps", return_value=[]
-    ):
+    with patch.object(hass, "config_entries") as mock_config_entries:
```

Comment:
We shouldn't patch the core interfaces. Instead check that a flow is either in progress or a new entry created, as many as configured. In this case I think a new entry will be created.

We should prefer to patch in the periphery when possible. Eg if the library would be involved we would patch the library.

To check if a flow is in progress we have this api: `hass.config_entries.flow.async_progress`.

https://github.com/home-assistant/home-assistant/blob/80a55360dc0cba19f2b7ccfdc84ea369ae368f73/homeassistant/data_entry_flow.py#L85-L92

Reply:
I rewrote the tests and check if entry is created or not.